### PR TITLE
Make Player Hand Lag Optional

### PR DESCRIPTION
**Pull request by joshuaskelly**
hand-lag-option-demo

## Summary
Many players complain about mouse acceleration, but there is no mouse acceleration. I believe they are mistaking hand lag for mouse acceleration. This change allows players to turn off hand lag.

## Changes
- Added a hand lag toggle option
- Added a `handLagStrength` field to the Player class to make the effect tunable.
- If hand lag toggled off, weapons don't move down when player looks up.
- Made offhand have parity with primary hand. Namely support for drawable meshes.
- Made held item offsets configurable on Player object.

### CODE REVIEW

**Review by PythooonUser — APPROVED** [review]
Looks good!

File: Dungeoneer/src/com/interrupt/dungeoneer/gfx/GlRenderer.java [hunk 1]
```diff
@@ -2017,8 +2017,12 @@ protected void drawHeldItem() {
 			if(handMaxLerpTime > 1) handMaxLerpTime = 1f;
 		}
 
-		handMaxDirection.lerp(handLagRotation, 0.6f);
-		forwardDirection.set(camera.direction).lerp(handLagRotation, 0.6f);
+		float handLagStrength = Game.instance.player.handLagStrength;
+		if (!Options.instance.handLagEnabled) {
+		    handLagStrength = 0;
+        }
```
Comment (PythooonUser) [inline Dungeoneer/src/com/interrupt/dungeoneer/gfx/GlRenderer.java:2023]: Formatting seems off.

**Review by PythooonUser — APPROVED** [review]
Looks good!

File: Dungeoneer/src/com/interrupt/dungeoneer/entities/Player.java [hunk 2]
```diff
@@ -73,6 +73,18 @@
...
 	public float headBobHeight = 0.3f;
 
+	/** Hand lag strength. */
+	public float handLagStrength = 0.6f;
+
+    /** Offhand lag strength. */
+	public float offhandLagStrength = 0.8f;
+
+	/** Hand offset */
+	public Vector3 handOffset = new Vector3(-0.12f, -0.07f, 0.28f);
```
Comment (Interrupt) [inline Dungeoneer/src/com/interrupt/dungeoneer/entities/Player.java:83]: Making  this configurable is great!

**Review by Interrupt — APPROVED** [review]

**Pull request merged**